gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -230,6 +232,29 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
     {% endif %}
     {% endfor %}
 
+{# TODO: Allow client creation to explictly specify connection
+         parameters, such as secure/insecure connections, and have tests use
+         that. Then remove this function.
+         Tracking issue:
+         https://github.com/googleapis/gapic-generator-python/issues/901
+#}
+def _protocol_for(host: str) -> str:
+    """Determine the URL protocol to use for connecting to the given host.
+
+    This function returns 'http' for hosts that match
+    `^localhost(:.*)?$` and `https` for all others. This is intended
```

Comment:
Do we have anything like this in the other languages? Overall this looks suspicious I would prefer to not have something like this. Determining protocol from host name does not seem reliable and also we are modifying production code for testing purposes. Also what if the host is specified as an IP (land specifically 127.0.0.1)?

Reply:
Yes. This was discussed in an internal doc; I'll ping you the link. We're specifically looking for `localhost` because yes, this is a temporary special case to allow testing (see https://github.com/googleapis/gapic-generator-python/issues/901 for the long-term solution)